### ISSUE TITLE
ALB-038: entrenar saves initialization weights to SafeTensors, not trained weights

### ISSUE THREAD

**Issue by noahgift** [paiml/albor#36]
## Summary

entrenar's SafeTensors checkpoint contains **initialization weights**, not the actual trained weights from GPU memory. Training reduces loss normally (10.3 → 4.42 over 31 steps in the 50M run), but the saved `model.safetensors` file contains identical, deterministic weight patterns across all layers.

## Evidence

1. **All 12 layers are byte-identical**: `model.layers.0.self_attn.q_proj.weight` through `model.layers.11.self_attn.q_proj.weight` contain the exact same bytes. Same for gate_proj, k_proj, etc.

2. **Norm weights are exactly 1.0**: All 25 LayerNorm weights (input_layernorm, post_attention_layernorm, model.norm) are precisely 1.0 with zero variance — never updated from initialization.

3. **Values are deterministic patterns, not random**: The first 14 values of q_proj form a smooth sinusoidal ramp (0.0, 0.0054, 0.0108, 0.0159, ..., 0.0442), not random initialization.

4. **Identical statistics across layers**: Layer 0 and Layer 11 q_proj have identical std (0.031250), skewness (-0.0001), and kurtosis (-1.5002).

5. **Perplexity matches random model**: Pure-Python inference on the saved checkpoint gives CE=10.94 (worse than random CE=10.40), despite training loss reporting 4.42.

## Impact

- **CRITICAL**: Blocks ALL model evaluation and downstream phases
- The currently-running 350M training (~20h on 4090) will produce an equally broken checkpoint
- ALB-037 (realizar ignores weights) may be a secondary effect — the weights were never properly saved
- All post-training phases (distillation, fine-tuning, merging, quantization) are blocked

## Expected Behavior

After training for 31 steps with loss 4.42:
- Each layer should have unique weights (different from initialization)
- Norm weights should have moved from 1.0
- Perplexity on training data should be ~83 (exp(4.42))

## Reproduction

```bash
# Train 50M model
apr train apply --task pretrain --config configs/train/pretrain-50m.yaml

# Check if layers are identical (they should NOT be)
python3 -c "
import json, struct, numpy as np
with open('checkpoints/albor-base-50m/model.safetensors', 'rb') as f:
    hs = struct.unpack('<Q', f.read(8))[0]
    h = json.loads(f.read(hs).decode())
    d = f.read()
q0 = d[h['model.layers.0.self_attn.q_proj.weight']['data_offsets'][0]:h['model.layers.0.self_attn.q_proj.weight']['data_offsets'][1]]
q1 = d[h['model.layers.1.self_attn.q_proj.weight']['data_offsets'][0]:h['model.layers.1.self_attn.q_proj.weight']['data_offsets'][1]]
print(f'Layer 0 == Layer 1: {q0 == q1}')  # Should print False
"
```

## Root Cause Hypothesis

entrenar likely creates the SafeTensors file during model initialization and either:
1. Never copies trained weights from GPU memory back to the file
2. Uses a weight serialization path that reads the initialization buffer instead of the current parameter values
3. Shares a single weight buffer for all layers during init and never differentiates them

## Relationship to Other Issues

- Subsumes ALB-037 (realizar weight loading) as a contributing factor
- Related to ALB-035 (no training_state.json) — both indicate incomplete checkpoint serialization
- Related to ALB-025 (no monitoring state) — pattern of missing state export
